x/mint/abci.go

```diff
@@ -0,0 +1,93 @@
+package mint
+
+import (
+	"time"
+
+	"github.com/celestiaorg/celestia-app/x/mint/keeper"
+	"github.com/celestiaorg/celestia-app/x/mint/types"
+	"github.com/cosmos/cosmos-sdk/telemetry"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// BeginBlocker updates the inflation rate, annual provisions, and then mints
+// the block provision for the current block.
+func BeginBlocker(ctx sdk.Context, k keeper.Keeper) {
+	defer telemetry.ModuleMeasureSince(types.ModuleName, time.Now(), telemetry.MetricKeyBeginBlocker)
+
+	maybeSetGenesisTime(ctx, k)
+	maybeUpdateMinter(ctx, k)
+	mintBlockProvision(ctx, k)
+	setPreviousBlockTime(ctx, k)
+}
+
+// maybeSetGenesisTime sets the genesis time if the current block height is 1.
+func maybeSetGenesisTime(ctx sdk.Context, k keeper.Keeper) {
+	if ctx.BlockHeight() == 1 {
+		genesisTime := ctx.BlockTime()
+		minter := k.GetMinter(ctx)
+		minter.GenesisTime = &genesisTime
+		k.SetMinter(ctx, minter)
+	}
+}
```

Comment:
I have an alternative proposal to this which you can do in a follow up PR if you choose to. Rather than this, add a line in:

https://github.com/celestiaorg/celestia-app/blob/1e60076cd52de601b6e403bc783389d17faa46ef/app/app.go#L571-L578

Something like:

```go
app.MintKeeper.SetGenesisTime(req.Time)
```
Which will set the genesis time. This also avoids the problem of `PreviousBlockTime` being nil at height 1